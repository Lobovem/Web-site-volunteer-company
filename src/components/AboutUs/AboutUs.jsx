import { TitleMain } from '../TitleMain/TitleMain';
import s from './AboutUs.module.scss';

export const AboutUs = () => {
  return (
    <section className={s.aboutUs}>
      <div className={s.aboutUs__contentWrap}>
        <div className={s.aboutUs__imgWrap}>
          <img className={s.aboutUs__img} src="/img/aboutUs.jpg" alt="aboutUs.jpg" />
        </div>
        <div className={s.aboutUs__textWrap}>
          <TitleMain title="UkrHumBat - команда людей, які діють" />

          <p className={s.aboutUs__desc}>
            Саме зараз час і місце, коли кожен громадянин своєї країни має стати на її захист і своїми справами: кожен на власному фронті
            має робити все для Перемоги. Тому дуже важливо, щоб і на передовій, і в тилу, проявлялася позиція громадянина, реалізовувалося
            глибоке розуміння громадянського обов'язку.
          </p>

          <h3 className={s.aboutUs__title}>Разом до перемоги</h3>
          <p className={s.aboutUs__desc}>
            Таке єднання фронту і тилу, що забезпечує служіння єдиній справі - загальній Перемозі українського народу над російськими
            загарбниками, забезпечують сьогодні волонтери нашого фонду.
          </p>
        </div>
      </div>
    </section>
  );
};
