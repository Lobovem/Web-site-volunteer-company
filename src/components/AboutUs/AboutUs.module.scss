@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';


.aboutUs {
  // background: $background-grey;

  &__wrap {

    @include layoutMain;
    // padding: 50px 0;

    // @include lg-min {
    //   padding: 120px 0;
    // }
  }

  &__contentWrap {
    @include lg-min {
      display: flex;
      justify-content: space-between;
      gap: 40px;
    }
  }

  &__textWrap {
    @include lg-min {
      flex-basis: 50%;
    }
  }

  &__img {
    border-radius: 15px;
    max-width: 100%;
    height: auto;
  }

  &__imgWrap {
    display: flex;
    justify-content: center;
    margin-bottom: 20px;

    @include lg-min {
      display: block;
      flex-basis: 50%;
    }
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-18;
    line-height: 18px;
    color: $color-black;
    mix-blend-mode: normal;

    @include lg-min {
      font-size: $font-size-24;
      line-height: 24px;
      margin-bottom: 30px;
    }
  }

  &__desc {
    font-size: $font-size-14;
    line-height: 21px;
    color: $color-black;
    mix-blend-mode: normal;

    @include lg-min {
      margin-bottom: 50px;
    }
  }
}