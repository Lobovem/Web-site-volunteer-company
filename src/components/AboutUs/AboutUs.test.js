import { render, screen } from '@testing-library/react';
import { AboutUs } from './AboutUs';

describe('AboutUs', () => {
  test('renders AboutUs component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(<AboutUs />);
    expect(component).toMatchSnapshot();
  });

  test('renders component without errors', () => {
    render(<AboutUs />);
    // You can add additional assertions if needed
  });

  test('should render the TitleMain component', () => {
    render(<AboutUs />);
    const titleMainComponent = screen.getByRole('heading', { name: /UkrHumBat - команда людей, які діють/i });
    expect(titleMainComponent).toBeInTheDocument();
  });

  test('renders the description paragraphs', () => {
    render(<AboutUs />);
    const description1 = screen.getByText(
      "Саме зараз час і місце, коли кожен громадянин своєї країни має стати на її захист і своїми справами: кожен на власному фронті має робити все для Перемоги. Тому дуже важливо, щоб і на передовій, і в тилу, проявлялася позиція громадянина, реалізовувалося глибоке розуміння громадянського обов'язку."
    );
    const description2 = screen.getByText(
      'Таке єднання фронту і тилу, що забезпечує служіння єдиній справі - загальній Перемозі українського народу над російськими загарбниками, забезпечують сьогодні волонтери нашого фонду.'
    );
    expect(description1).toBeInTheDocument();
    expect(description2).toBeInTheDocument();
  });

  test('search tag h2', () => {
    render(<AboutUs />);
    const h2Element = screen.getByRole('heading', { level: 2 });
    expect(h2Element).toBeInTheDocument();
  });

  test('search tag h3', () => {
    render(<AboutUs />);
    const h3Element = screen.getByRole('heading', { level: 3 });
    expect(h3Element).toBeInTheDocument();
  });

  test('search the image', () => {
    render(<AboutUs />);
    const imgElement = screen.getByRole('img');
    expect(imgElement).toBeInTheDocument();
  });
});
