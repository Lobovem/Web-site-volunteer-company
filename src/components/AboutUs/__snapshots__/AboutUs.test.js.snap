// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs renders AboutUs component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="aboutUs"
      >
        <div
          class="aboutUs__contentWrap"
        >
          <div
            class="aboutUs__imgWrap"
          >
            <img
              alt="aboutUs.jpg"
              class="aboutUs__img"
              src="/img/aboutUs.jpg"
            />
          </div>
          <div
            class="aboutUs__textWrap"
          >
            <h2
              class="titleMain"
            >
              UkrHumBat - команда людей, які діють
            </h2>
            <p
              class="aboutUs__desc"
            >
              Саме зараз час і місце, коли кожен громадянин своєї країни має стати на її захист і своїми справами: кожен на власному фронті має робити все для Перемоги. Тому дуже важливо, щоб і на передовій, і в тилу, проявлялася позиція громадянина, реалізовувалося глибоке розуміння громадянського обов'язку.
            </p>
            <h3
              class="aboutUs__title"
            >
              Разом до перемоги
            </h3>
            <p
              class="aboutUs__desc"
            >
              Таке єднання фронту і тилу, що забезпечує служіння єдиній справі - загальній Перемозі українського народу над російськими загарбниками, забезпечують сьогодні волонтери нашого фонду.
            </p>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="aboutUs"
    >
      <div
        class="aboutUs__contentWrap"
      >
        <div
          class="aboutUs__imgWrap"
        >
          <img
            alt="aboutUs.jpg"
            class="aboutUs__img"
            src="/img/aboutUs.jpg"
          />
        </div>
        <div
          class="aboutUs__textWrap"
        >
          <h2
            class="titleMain"
          >
            UkrHumBat - команда людей, які діють
          </h2>
          <p
            class="aboutUs__desc"
          >
            Саме зараз час і місце, коли кожен громадянин своєї країни має стати на її захист і своїми справами: кожен на власному фронті має робити все для Перемоги. Тому дуже важливо, щоб і на передовій, і в тилу, проявлялася позиція громадянина, реалізовувалося глибоке розуміння громадянського обов'язку.
          </p>
          <h3
            class="aboutUs__title"
          >
            Разом до перемоги
          </h3>
          <p
            class="aboutUs__desc"
          >
            Таке єднання фронту і тилу, що забезпечує служіння єдиній справі - загальній Перемозі українського народу над російськими загарбниками, забезпечують сьогодні волонтери нашого фонду.
          </p>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
